Automattic/wp-calypso
# Videos UI: Accept a full footer component as a prop.

This is the footer version of #58434, moving the footer out of `<VideosUi />` to its own component, that can then be passed in as a prop. This allows maximum versatility for the footer, while doing away with most of the list of props in the current version.

This PR also fixes three outstanding footer-related bugs.

Fixes #58207, fixes #58366, fixes #58184.

| Production | This PR |
| --- | --- |
| <img width="443" alt="Screen Shot 2021-11-24 at 12 56 34 PM" src="https://user-images.githubusercontent.com/349751/143311806-4a9a9083-2871-44d6-b9da-c3b6b2efe525.png"> | <img width="414" alt="Screen Shot 2021-11-24 at 12 55 57 PM" src="https://user-images.githubusercontent.com/349751/143311817-c900a20b-7841-4c92-987e-09e935f0be75.png"> |

**Testing Instructions**
* Load this PR locally.
* Open the videos UI.
* Verify the footer displays with no console errors.
* Verify there's no link to draft a post for mobile.
* Verify the footer is fixed for mobile.
* Verify there's no footer border at desktop widths.
